tests/test_web_runner.py

```diff
@@ -211,3 +211,22 @@ async def mock_create_server(*args, **kwargs):
     assert server is runner.server
     assert host is None
     assert port == 8080
+
+
+def test_run_after_asyncio_run() -> None:
+    async def nothing():
+        pass
+
+    async def shutdown():
+        raise web.GracefulExit()
```

Comment:
Or maybe something like this:
```suggestion
    class Spy:
        called_spy = False
        @classmethod
        def spy_on(cls):
            cls.called_spy = True

    async def shutdown():
        Spy.spy_on()
        raise web.GracefulExit()
```

And then add this at the end:
```python
assert Spy.called_spy, "run_app() should work after asyncio.run()."
```